osu.Game.Rulesets.Catch/Skinning/Legacy/CatchLegacySkinTransformer.cs

```diff
@@ -22,14 +24,22 @@ public CatchLegacySkinTransformer(ISkinSource source)
 
         public override Drawable GetDrawableComponent(ISkinComponent component)
         {
-            if (component is HUDSkinComponent hudComponent)
+            if (component is SkinnableTargetComponent targetComponent && targetComponent.Target == SkinnableTarget.MainHUDComponents)
             {
-                switch (hudComponent.Component)
+                if (!providesComboCounter)
+                    return null;
+
+                if (Source.GetDrawableComponent(component) is SkinnableTargetComponentsContainer components)
                 {
-                    case HUDSkinComponents.ComboCounter:
-                        // catch may provide its own combo counter; hide the default.
-                        return providesComboCounter ? Drawable.Empty() : null;
+                    // catch may provide its own combo counter; hide the default.
+                    // todo: this should be done in an elegant way per ruleset, defining which HUD skin components should be displayed.
+                    foreach (var legacyComboCounter in components.OfType<LegacyComboCounter>())
+                        legacyComboCounter.ContentVisible = false;
+
+                    return components;
                 }
+
+                return null;
```

Comment:
Why can't this be applied like it was in the old code? Why does it need to be done on the children of the `MainHUDComponents` lookup?

Reply:
Because of the case where the `MainHUDComponents` target may have been customised, the combo counter will then be instantiated via reflection instead rather than `GetDrawableComponent`.

Reply:
This was the missing information I was looking for which wasn't mentioned anywhere.

Seems like `HUDSkinComponent` in general should just be removed to avoid confusion?

Reply:
Hmm, can't see any problem with that, will do it.